### PR TITLE
MS16206: flying fix (RC70.1)

### PR DESCRIPTION
**Pull request by r3tk0n**
Addresses Manuscript [#16206](https://highfidelity.manuscript.com/f/cases/16206). Flying should always be enabled in desktop mode (unless overridden by a script or something).

Flying should be off for users who launch in HMD on their first run.

The "Flying & jumping" menu item has been removed from the "Controls..." menu in desktop mode, but remains if accessed via the tablet in VR mode.

### CODE REVIEW

**Comment by r3tk0n** [comment]
Test Plan:

Goal: Verify the following:
1.) Flying is now enabled by default for desktop users.
2.) "Flying & jumping" menu item has been removed from "Settings > Controls..." menu in desktop mode.
3.) "Flying & jumping" menu item remains in "Tablet > Menu > Settings > Controls..." menu in HMD mode.

Requirements: HMD, fresh install with no settings stored (purge %appdata% HiFi directories).

Steps:

- Start Interface.exe in desktop mode.
- Attempt to fly. If successful, Goal 1 is good.
- Navigate to "Settings > Controls..." in desktop mode
- If "Flying & jumping" checkbox is not listed, Goal 2 is good.
- Close Interface.exe and purge settings.
- Start Interface.exe in HMD mode.
- Navigate to "Tablet > Menu > Settings > Controls..." menu via tablet in HMD mode.
- If "Flying & jumping" checkbox is present then goal 3 is good.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11463/HighFidelity-Beta-PR13662-41fbf4c.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11463/HighFidelity-Beta-PR13662-41fbf4c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11463/HighFidelity-Beta-Interface-PR13662-41fbf4c.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11463/HighFidelity-Beta-PR13662-41fbf4c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11463/HighFidelity-Beta-Interface-PR13662-41fbf4c.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4830,7 +4830,9 @@ void Application::loadSettings() {
         // should be enabled by default in desktop mode.
 
         auto myAvatar = getMyAvatar();
-        myAvatar->setFlyingEnabled((isFirstPerson)?false:true);
+        bool initialFlyingVal = !isFirstPerson;
+        myAvatar->setFlyingEnabled(initialFlyingVal);
+        Setting::Handle<bool> initFlyingSetting {"enabledFlying", initialFlyingVal};
```
Comment (wayne-chen) [inline interface/src/Application.cpp]: This is unused. I think this line should be removed.

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -1461,7 +1461,7 @@ private slots:
     std::array<float, MAX_DRIVE_KEYS> _driveKeys;
     std::bitset<MAX_DRIVE_KEYS> _disabledDriveKeys;
 
-    bool _enableFlying { false };
+    bool _enableFlying { true };
```
Comment (wayne-chen) [inline interface/src/avatar/MyAvatar.h]: Why does this need to be set to `true`?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11470/HighFidelity-Beta-PR13662-92bc10f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11470/HighFidelity-Beta-PR13662-92bc10f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11470/HighFidelity-Beta-Interface-PR13662-92bc10f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11470/HighFidelity-Beta-PR13662-92bc10f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11470/HighFidelity-Beta-Interface-PR13662-92bc10f.exe

**Comment by r3tk0n** [comment]
Should be good to go, same test plan as above.

**Comment by qlimguicheteau** [comment]
@conklin94122 
PASS

PASS-- Flying is enabled for desktop mode on fresh install.
PASS-- No checkbox in desktop mode
PASS-- Flying and Jumping visible in HMD

Cleared settings works for both startup attempts. Tansitioning between the two also works, menus remain accurate both ways.

**Pull request merged**